controllers/actions.github.com/ephemeralrunner_controller.go

```diff
@@ -80,6 +81,40 @@ func (r *EphemeralRunnerReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			return ctrl.Result{}, nil
 		}
 
+		if controllerutil.ContainsFinalizer(ephemeralRunner, ephemeralRunnerActionsFinalizerName) {
+			if ephemeralRunner.Status.Phase == corev1.PodSucceeded {
```

Comment:
this feels like a weak assumption.

Reply:
add a new `Deleted` to `ephemeralRunner.Status` or something to indicate it has been deleted?

Reply:
So the PodSucceeded phase can only be set in case we checked the pod, checked with the service and everything is fine. Otherwise, ephemeralRunner can never reach the PodSucceeded phase. That is why I wanted to avoid another check since we are checking before we are setting the phase to this one